GamepadEventDrivenInputAPI/explainer.md

```diff
@@ -0,0 +1,251 @@
+
+These limitations make it challenging to deliver consistent, low-latency input experiences in the browser. An event-driven Gamepad API (similar to existing keyboard and mouse event models) would allow applications to respond immediately to input changes as they occur, reducing the reliance on polling and enabling real-time responsiveness for latency-critical use cases.
+
+### Developer code sample of existing poll based API
+```JS
+function pollGamepadInput() {
+  const gamepads = navigator.getGamepads();
+
+  for (const gamepad of gamepads) {
+    if (!gamepad) continue;
+    // Example: Logging the first axis and button.
+    const axisX = gamepad.axes[0];
+    const buttonA = gamepad.buttons[0].pressed;
+
+    console.log(`Axis X: ${axisX}, Button A pressed: ${buttonA}`);
+  }
+
+  // Continue polling in the next animation frame.
+  requestAnimationFrame(pollGamepadInput);
+}
+
+// Start polling.
+window.addEventListener('gamepadconnected', () => {
+  console.log('Gamepad connected!');
+  requestAnimationFrame(pollGamepadInput);
+}); 
+```
+#### Key Points:
+- navigator.getGamepads() returns a snapshot of all connected gamepads.
+- The polling loop is driven by `requestAnimationFrame`, typically around 60Hz (matching display refresh rate), which is much lower than the internal OS poll rate (eg., 250Hz). This mismatch can result in missed input updates, making the 60Hz rate insufficient for latency-critical applications like cloud gaming.
+
+## Goals
+
+Reduce input latency by moving away from constant polling and introducing event-driven input handling. 
+
+## Non-goals
+
+The existing polling mechanism will not be deprecated. We are just proposing an alternative way of handling input events and applications are free to select whichever they prefer.
+
+## Proposed Approach
+To address the challenges of input latency, this proposal introduces a new event-driven mechanism: The `rawgamepadinputchange` event. This event fires directly on the [Gamepad](https://w3c.github.io/gamepad/#dom-gamepad) object and delivers real-time updates for each input frame, eliminating the need for high-frequency polling. The `rawgamepadinputchange` event includes detailed information about the state of the gamepad at the moment of change:
+
+- `axesChanged` and `buttonsChanged`: Arrays of indices indicating which axes or button values changed since the last event.
+
+- `buttonsPressed` and `buttonsReleased`: Indices of buttons whose pressed state transitioned (from pressed to released or vice versa).
+
+- `gamepadSnapshot`: A complete snapshot of the gamepad's current state, including all axes, buttons, ID, index, and timestamp.
```

Comment:
We might want to explain how `gamepadSnapshot` is a different object than the `rawgamepadinputchange` event's target property:

https://developer.mozilla.org/en-US/docs/Web/API/Event/target

The event target is the live object where the `gamepadSnapshot` is frozen at the event's input frame.